src/object/schema_system_catalog_builder.cpp

```diff
@@ -111,26 +129,36 @@ namespace cubschema
       {
 	DB_CONSTRAINT_TYPE type = c.type;
 
-	const char *name = c.name.data();
+	const char *name = nullptr;
+	if (!c.name.empty ())
+	  {
+	    name = c.name.data();
+	  }
+
 	int is_class_attribute = c.is_class_attributes ? 1 : 0;
-	if (type == DB_CONSTRAINT_INDEX)
+
+	switch (type)
+	  {
+	  case DB_CONSTRAINT_INDEX:
+	  case DB_CONSTRAINT_PRIMARY_KEY:
+	  case DB_CONSTRAINT_UNIQUE:
+	  case DB_CONSTRAINT_NOT_NULL:
```

Comment:
In the case of DB_CONSTRAINT_NOT_NULL, there was a check using do_check_rows_for_null() by calling the db_constrain_non_null() function, but is this no longer necessary?

Reply:
Because this routine is creating a new system table, IMO, there is no reason to check that rows in the table has null. Because, in this routine, every constraint is added in empty system table, do_check_rows_for_null () performs `SELECT count(*) FROM tbl WHERE att IS NULL` unnecessarily.